### PR TITLE
Improve kubeadm documentation

### PR DESCRIPTION
**Pull request by luxas**
> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> For 1.9 Features: set Milestone to `1.9` and Base Branch to `release-1.9`
> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> NOTE: Please check the “Allow edits from maintainers” box (see image below) to
> [allow reviewers to fix problems](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) on your patch and speed up the review process.
>
> Please delete this note before submitting the pull request.
>
> NOTE: After opening the PR, please *un-check and re-check* the "Allow edits from maintainers" box. This is a temporary workaround to address a known issue with GitHub.

Allow edits from maintainers checkbox

Edits, updates and improvements to the existing kubeadm docs.
cc @kubernetes/sig-cluster-lifecycle-pr-reviews

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/website/6645)

### CODE REVIEW

File: docs/reference/setup-tools/kubeadm/kubeadm-token.md [hunk 1]
```diff
@@ -6,11 +6,13 @@ approvers:
 title: kubeadm token
 ---
 {% capture overview %}
+
 Bootstrap tokens are used for establishing bidirectional trust between a node joining 
 the cluster and a the master node, as described in [authenticating with bootstrap tokens](/docs/admin/bootstrap-tokens/).
 
-`kubeadm-init` creates an initial token with 24h TTL. The following commands allow you to manage 
+`kubeadm-init` creates an initial token with a 24-hour TTL. The following commands allow you to manage
```
Comment (jbeda) [inline docs/reference/setup-tools/kubeadm/kubeadm-token.md]: Should there be a dash in `kubeadm init`?

File: docs/reference/setup-tools/kubeadm/kubeadm-alpha.md [hunk 2]
```diff
@@ -118,7 +119,7 @@ You can create all required kubeconfig files with the `all` sub command, or sele
 
 ## kubeadm alpha phase controlplane {#cmd-phase-controlplane}
 
-You can create all required static pod files for control plane components with the `all` sub command, 
+You can create all required static Pod files for the control plane components with the `all` subcommand,
```
Comment (timothysc) [inline docs/reference/setup-tools/kubeadm/kubeadm-alpha.md]: I don't see why you are capitalizing Pod all over unless it is that way in the rest of the docs.?.?

/cc @Bradamant3

File: docs/reference/setup-tools/kubeadm/kubeadm-alpha.md [hunk 3]
```diff
@@ -205,8 +206,8 @@ or selectively configure single elements.
 
 ## kubeadm alpha phase upload-config {#cmd-phase-upload-config}
 
-You can use this command to upload the configuration of your cluster. Alternatively, you 
-can use [kubeadm config](kubeadm-config.md).
+You can use this command to upload the kubeadm-specific configuration/desired state of your cluster.
```
Comment (timothysc) [inline docs/reference/setup-tools/kubeadm/kubeadm-alpha.md]: "You can use this command to upload the kubeadm configuration to your cluster."

File: docs/reference/setup-tools/kubeadm/kubeadm-init.md [hunk 4]
```diff
@@ -241,7 +244,7 @@ Here's a breakdown of what/why:
    `https://{node-ip}:10250/stats/`. If you want to enable cAdvisor to listen on a
    wide-open port, run:
 
-   ```
+   ```bash
```
Comment (timothysc) [inline docs/reference/setup-tools/kubeadm/kubeadm-init.md]: this doesn't seem necessary

File: docs/reference/setup-tools/kubeadm/kubeadm-init.md [hunk 5]
```diff
@@ -252,12 +255,11 @@ Here's a breakdown of what/why:
...
-Since the [Kubernetes 1.6 release](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG.md#node-components-1),
-Kubernetes has started using CRI as the default container runtime.
-The container runtime used by kubeadm is Docker, which is enabled through the built-in
-`dockershim` in `kubelet` mechanisms.
+Since v1.6.0, Kubernetes has enabled the use of CRI, Container Runtime Interface, by default.
+The container runtime used by default is Docker, which is enabled through the built-in
+`dockershim` CRI implementation inside of the `kubelet`.
 
-Other CRI-based runtimes include::
+Well-known, external CRI-based runtimes include:
```
Comment (timothysc) [inline docs/reference/setup-tools/kubeadm/kubeadm-init.md]: I liked the previous statement with a single `:`

File: docs/reference/setup-tools/kubeadm/kubeadm-join.md [hunk 6]
```diff
@@ -49,7 +52,7 @@ openssl x509 -pubkey -in /etc/kubernetes/pki/ca.crt | openssl rsa -pubin -outfor
 
 **Example `kubeadm join` command:**
 
-```
+```bash
```
Comment (timothysc) [inline docs/reference/setup-tools/kubeadm/kubeadm-join.md]: same

File: docs/setup/independent/troubleshooting-kubeadm.md [hunk 7]
```diff
@@ -32,7 +32,7 @@ If you see the following warnings while running `kubeadm init`
 [preflight] WARNING: ethtool not found in system path                           
 ```
 
-Then you may be missing ebtables and ethtool on your Linux machine. You can install them with the following commands: 
+Then you may be missing `ebtables`, `ethtool` or a similar executable on your Linux machine. You can install them with the following commands:
```
Comment (timothysc) [inline docs/setup/independent/troubleshooting-kubeadm.md]: what is "similar"?

File: docs/reference/setup-tools/kubeadm/kubeadm-alpha.md (hunk 2 above)
Comment (zacharysarah) [reply docs/reference/setup-tools/kubeadm/kubeadm-alpha.md]: @timothysc It's the correct convention, yes.

File: docs/reference/setup-tools/kubeadm/kubeadm-init.md (hunk 4 above)
Comment (zacharysarah) [reply docs/reference/setup-tools/kubeadm/kubeadm-init.md]: It's helpful.

File: docs/reference/setup-tools/kubeadm/kubeadm-alpha.md (hunk 3 above)
Comment (zacharysarah) [reply docs/reference/setup-tools/kubeadm/kubeadm-alpha.md]: @luxas ☝️

File: docs/reference/setup-tools/kubeadm/kubeadm-init.md (hunk 5 above)
Comment (zacharysarah) [reply docs/reference/setup-tools/kubeadm/kubeadm-init.md]: @luxas ☝️

File: docs/reference/setup-tools/kubeadm/kubeadm-join.md (hunk 6 above)
Comment (zacharysarah) [reply docs/reference/setup-tools/kubeadm/kubeadm-join.md]: Syntax-specific formatting is helpful.

File: docs/setup/independent/troubleshooting-kubeadm.md (hunk 7 above)
Comment (zacharysarah) [reply docs/setup/independent/troubleshooting-kubeadm.md]: I'm fine with this as is.

File: docs/reference/setup-tools/kubeadm/kubeadm-init.md [hunk 8]
```diff
@@ -185,11 +184,15 @@ use case. This means you can, for example, copy an existing CA into `/etc/kubern
...
 of the certs.
 
+#### External CA mode {#external-ca-mode}
+
 It is also possible to provide just the `ca.crt` file and not the 
 `ca.key` file (this is only available for the root CA file, not other cert pairs). 
 If all other certificates and kubeconfig files are in place, kubeadm recognizes 
-this condition and activates the "ExternalCA" mode, which also
-implies the CSR signer controller in the controller manager won't be started.
+this condition and activates the "External CA" mode, which means that kubeadm will
```
Comment (zacharysarah) [inline docs/reference/setup-tools/kubeadm/kubeadm-init.md]: Replace through the end of the paragraph with:
```
this condition and activates the "External CA" mode. kubeadm will proceed without the CA key on disk.

**Note:** Don't run the CSR signer controller. Instead, run the controller-manager standalone with `--controllers=csrsigner` and point to the CA certificate and key.
{: .note}
```
So:
> this condition and activates the "External CA" mode. kubeadm will proceed without the CA key on disk.
>
> **Note:** Don't run the CSR signer controller. Instead, run the controller-manager standalone with `--controllers=csrsigner` and point to the CA certificate and key.
> {: .note}

File: docs/reference/setup-tools/kubeadm/kubeadm-token.md (hunk 1 above)
Comment (luxas) [reply docs/reference/setup-tools/kubeadm/kubeadm-token.md]: fixed

File: docs/reference/setup-tools/kubeadm/kubeadm-alpha.md (hunk 3 above)
Comment (luxas) [reply docs/reference/setup-tools/kubeadm/kubeadm-alpha.md]: fixed

File: docs/reference/setup-tools/kubeadm/kubeadm-init.md (hunk 5 above)
Comment (luxas) [reply docs/reference/setup-tools/kubeadm/kubeadm-init.md]: fixed

File: docs/reference/setup-tools/kubeadm/kubeadm-init.md (hunk 8 above)
Comment (luxas) [reply docs/reference/setup-tools/kubeadm/kubeadm-init.md]: fixed

**Comment by luxas** [comment]
@zacharysarah Changes addressed.

**Comment by zacharysarah** [comment]
LGTM :shipit:

File: docs/reference/setup-tools/kubeadm/kubeadm-alpha.md [hunk 9]
```diff
@@ -220,7 +221,7 @@ can use [kubeadm config](kubeadm-config.md).
 
 ## kubeadm alpha phase addon {#cmd-phase-addon}
 
-You can install all the available addons with the `all` sub command, or 
+You can install all the available addons with the `all` subcommand, or 
 install them selectively.
```
Comment (fabriziopandini) [inline docs/reference/setup-tools/kubeadm/kubeadm-alpha.md]: Add something related to CoreOS?
Add something related to SupportIPVSProxyMode?

File: docs/reference/setup-tools/kubeadm/kubeadm-init.md [hunk 10]
```diff
@@ -44,11 +38,16 @@ following steps:
    Static Pod manifests are written to `/etc/kubernetes/manifests`; the kubelet
    watches this directory for Pods to create on startup.
 
-   Once control plane Pods are up and running, the kubeadm init sequence can continue.
+   Once control plane Pods are up and running, the `kubeadm init` sequence can continue.
```
Comment (fabriziopandini) [inline docs/reference/setup-tools/kubeadm/kubeadm-init.md]: Add something about dynamic kubelet configuration?

File: docs/reference/setup-tools/kubeadm/kubeadm-init.md [hunk 11]
```diff
@@ -185,11 +184,16 @@ use case. This means you can, for example, copy an existing CA into `/etc/kubern
...
 It is also possible to provide just the `ca.crt` file and not the 
 `ca.key` file (this is only available for the root CA file, not other cert pairs). 
 If all other certificates and kubeconfig files are in place, kubeadm recognizes 
-this condition and activates the "ExternalCA" mode, which also
-implies the CSR signer controller in the controller manager won't be started.
+this condition and activates the "External CA" mode. kubeadm will proceed without the
+CA key on disk.
+
+Instead, run the controller-manager standalone with `--controllers=csrsigner` and
+point to the CA certificate and key.
```
Comment (fabriziopandini) [inline docs/reference/setup-tools/kubeadm/kubeadm-init.md]: I'm not sure if this is correct. 
"Please note that in "External CA" mode, the csrsigning controller will be disabled."

File: docs/reference/setup-tools/kubeadm/kubeadm-upgrade.md [hunk 12]
```diff
@@ -6,6 +6,19 @@ approvers:
...
+for both planning an upgrade and actually performing it. `kubeadm upgrade` can also be used for downgrading
+cluster if necessary.
+
+Every upgrade process might be a bit different, so we've documented each minor upgrade process individually.
+Please check these documents out for more detailed how-to-upgrade guidance:
+
+ * [1.6 to 1.7 upgrades](/docs/tasks/administer-cluster/kubeadm-upgrade-1-7/)
+ * [1.7.x to 1.7.y  upgrades](/docs/tasks/administer-cluster/kubeadm-upgrade-1-8/)
+ * [1.7 to 1.8 upgrades](/docs/tasks/administer-cluster/kubeadm-upgrade-1-8/)
+ * [1.8.x to 1.8.y upgrades](/docs/tasks/administer-cluster/kubeadm-upgrade-1-8/)
```
Comment (fabriziopandini) [inline docs/reference/setup-tools/kubeadm/kubeadm-upgrade.md]: IT seems that this list doesn't show up in the preview...
What about the 1.8 to 1.9  upgrade?

File: docs/setup/independent/install-kubeadm.md [hunk 13]
```diff
@@ -4,19 +4,27 @@ title: Installing kubeadm
 
 {% capture overview %}
 
-This page shows how to use install kubeadm.
+This page shows how to use install the `kubeadm` toolbox.
```
Comment (fabriziopandini) [inline docs/setup/independent/install-kubeadm.md]: utility?

File: docs/reference/setup-tools/kubeadm/kubeadm-init.md (hunk 11 above)
Comment (luxas) [reply docs/reference/setup-tools/kubeadm/kubeadm-init.md]: That is correct

File: docs/setup/independent/install-kubeadm.md (hunk 13 above)
Comment (luxas) [reply docs/setup/independent/install-kubeadm.md]: We/I like to call it a toolbox :)

File: docs/reference/setup-tools/kubeadm/kubeadm-upgrade.md (hunk 12 above)
Comment (luxas) [reply docs/reference/setup-tools/kubeadm/kubeadm-upgrade.md]: rebase conflict -- please add it if you get to it

File: docs/setup/independent/create-cluster-kubeadm.md [hunk 14]
```diff
@@ -9,59 +9,73 @@ title: Using kubeadm to Create a Cluster
...
-* On GCE, [Google Kubernetes Engine](https://cloud.google.com/kubernetes-engine/)
-  gives you one-click Kubernetes clusters.
-* On Microsoft Azure, [Azure Container Service (AKS)](https://docs.microsoft.com/en-us/azure/aks/intro-kubernetes)
-  gives you managed Kubernetes clusters as a service.
-* On AWS, [kops](https://github.com/kubernetes/kops) makes cluster installation
-  and management easy.  kops supports building high availability clusters (a
-  feature that kubeadm is currently lacking but is building toward).
+**kubeadm** is a toolkit that help you bootstrap a best-practice Kubernetes
+cluster in an easy, reasonably secure and extensible way. It also supports
+managing [Bootstrap Tokens](#TODO) for you and upgrading/downgrading clusters.
```
Comment (fabriziopandini) [inline docs/setup/independent/create-cluster-kubeadm.md]: link missing

File: docs/setup/independent/create-cluster-kubeadm.md [hunk 15]
```diff
@@ -9,59 +9,73 @@ title: Using kubeadm to Create a Cluster
...
+
+kubeadm aims to set up a minimum viable cluster that pass the
+[Kubernetes Conformance tests](#TODO), but installing other addons than
```
Comment (fabriziopandini) [inline docs/setup/independent/create-cluster-kubeadm.md]: Link missing

File: docs/reference/setup-tools/kubeadm/kubeadm-init.md (hunk 10 above)
Comment (luxas) [reply docs/reference/setup-tools/kubeadm/kubeadm-init.md]: sure -- feel free to! I was just improving the existing docs, we might wanna add some new, shiny v1.9 stuff as well

File: docs/reference/setup-tools/kubeadm/kubeadm-alpha.md (hunk 9 above)
Comment (luxas) [reply docs/reference/setup-tools/kubeadm/kubeadm-alpha.md]: What would be CoreOS-specific?
We don't support that feature gate anymore

File: docs/setup/independent/create-cluster-kubeadm.md [hunk 16]
```diff
@@ -9,59 +9,73 @@ title: Using kubeadm to Create a Cluster
...
+really necessary for a functional cluster is out of scope.
+
+It by design does not install a networking solution for you, which means you
+have to install a third-party CNI-compliant networking solution yourself
+using `kubectl apply`.
+
+kubeadm expects the user to bring a machine to execute on, the type doesn't
```
Comment (fabriziopandini) [inline docs/setup/independent/create-cluster-kubeadm.md]: to provide?

File: docs/setup/independent/create-cluster-kubeadm.md (hunk 14 above)
Comment (luxas) [reply docs/setup/independent/create-cluster-kubeadm.md]: please fix those as follow-ups

File: docs/setup/independent/create-cluster-kubeadm.md (hunk 15 above)
Comment (luxas) [reply docs/setup/independent/create-cluster-kubeadm.md]: please fix those as follow-ups

File: docs/setup/independent/create-cluster-kubeadm.md (hunk 16 above)
Comment (luxas) [reply docs/setup/independent/create-cluster-kubeadm.md]: either is fine my me I guess...

File: docs/setup/independent/create-cluster-kubeadm.md [hunk 17]
```diff
@@ -9,59 +9,73 @@ title: Using kubeadm to Create a Cluster
...
+matter, can be a Linux laptop, virtual machine, physical/cloud server or
+Raspberry Pi. This makes kubeadm well suited to integrate with provisioning
+systems of different kinds (e.g. Terraform, Ansible, etc.).
+
+kubeadm is designed to be a good way for new users to start trying
```
Comment (fabriziopandini) [inline docs/setup/independent/create-cluster-kubeadm.md]: to be a simple way?

File: docs/setup/independent/create-cluster-kubeadm.md [hunk 18]
```diff
@@ -9,59 +9,73 @@ title: Using kubeadm to Create a Cluster
...
+Kubernetes out, possibly for the first time, an way for existing users to
```
Comment (fabriziopandini) [inline docs/setup/independent/create-cluster-kubeadm.md]: and a way for

File: docs/setup/independent/create-cluster-kubeadm.md [hunk 19]
```diff
@@ -9,59 +9,73 @@ title: Using kubeadm to Create a Cluster
...
+test their application on and stich together a cluster easily and to be
+a building block in a larger ecosystem and/or installer tool with a larger
+scope.
```
Comment (fabriziopandini) [inline docs/setup/independent/create-cluster-kubeadm.md]: ... users to test their applications. Kubeadm can be used also as a building block to be used in other installer or in higher level tools

File: docs/setup/independent/create-cluster-kubeadm.md [hunk 20]
```diff
@@ -9,59 +9,73 @@ title: Using kubeadm to Create a Cluster
...
-
-**Be sure to read the [limitations](#limitations)**.  Specifically, configuring
-cloud providers is difficult.
+| Area            | Maturity Level |
+|-----------------|--------------- |
+| Command line UX | beta           |
+| Implementation  | beta           |
+| Config file API | alpha          |
+| Self-hosting    | alpha          |
+| `kubeadm alpha` | alpha          |
```
Comment (fabriziopandini) [inline docs/setup/independent/create-cluster-kubeadm.md]: Dynamic kubelet config?
Core DNS?
others feautre under gate?

File: docs/setup/independent/create-cluster-kubeadm.md [hunk 21]
```diff
@@ -9,59 +9,73 @@ title: Using kubeadm to Create a Cluster
...
+### Support timeframes
+
+Kubernetes releases are generally supported for nine months, and during that
+period a patch release may be issued from the release branch if a severe bug or
+security issue is found. Here are the latest Kubernetes releases and the support
+timeframe; which also applies to `kubeadm`.
+
+| Kubernetes version | Release date | End-of-life-month |
+|--------------------|--------------|-------------------|
+| v1.6.x             | TODO         | December 2017     |
```
Comment (fabriziopandini) [inline docs/setup/independent/create-cluster-kubeadm.md]: TODO to be fixed

File: docs/setup/independent/create-cluster-kubeadm.md [hunk 22]
```diff
@@ -9,59 +9,73 @@ title: Using kubeadm to Create a Cluster
...
+kubeadm is designed to be a good way for new users to start trying
+Kubernetes out, possibly for the first time, an way for existing users to
+test their application on and stich together a cluster easily and to be
+a building block in a larger ecosystem and/or installer tool with a larger
+scope.
+
+You can install _kubeadm_ very easily on operating systems that support
+installing deb or rpm packages. The responsible SIG for kubeadm,
+[SIG Cluster Lifecycle](#TODO), provides these packages pre-built for you,
+but you may also on other OSes.
```
Comment (fabriziopandini) [inline docs/setup/independent/create-cluster-kubeadm.md]: Why kubeadm in italic?
Link to the kubeadm install doc?

File: docs/setup/independent/create-cluster-kubeadm.md [hunk 23]
```diff
@@ -408,102 +425,56 @@ scp root@<master ip>:/etc/kubernetes/admin.conf .
...
-``` 
-
-See 
-[their documentation](https://github.com/coreos/etcd/tree/master/etcdctl) for more 
-information.
+More options and information about the
+[`kubeadm reset command`](/docs/reference/setup-tools/kubeadm/kubeadm-reset/).
 
-## Upgrading
+## Upgrading a kubeadm cluster {#upgrades}
```
Comment (fabriziopandini) [inline docs/setup/independent/create-cluster-kubeadm.md]: 1.8->1.9

File: docs/setup/independent/create-cluster-kubeadm.md [hunk 24]
```diff
@@ -512,32 +483,31 @@ Instructions for upgrading kubeadm clusters are available for:
...
+## Feedback {#feedback}
 
 * kubeadm support Slack Channel:
-  [kubeadm](https://kubernetes.slack.com/messages/kubeadm/)
+  [#kubeadm](https://kubernetes.slack.com/messages/kubeadm/)
 * General SIG Cluster Lifecycle Development Slack Channel:
-  [sig-cluster-lifecycle](https://kubernetes.slack.com/messages/sig-cluster-lifecycle/)
-* Mailing List:
+  [#sig-cluster-lifecycle](https://kubernetes.slack.com/messages/sig-cluster-lifecycle/)
+* SIG Cluster Lifecycle [SIG information](#TODO)
```
Comment (fabriziopandini) [inline docs/setup/independent/create-cluster-kubeadm.md]: Link to be fixed

**Pull request merged**